File: data/EXP01/go/2467/2467.output_1.go
type Node struct {
    id     int
    parent int
    child  []int
}

func mostProfitablePath(edges [][]int, bob int, amount []int) int {
    n := len(amount)
    nodes := make([]Node, n)
    for _, edge := range edges {
        u, v := edge[0], edge[1]
        nodes[u].child = append(nodes[u].child, v)
        nodes[v].parent = u
    }

    dfs(&nodes, bob, amount, math.MinInt64)
    return nodes[bob].profit
}

func dfs(nodes *[]Node, curr int, amount []int, maxProfit int) {
    node := &((*nodes)[curr])
    gate := amount[curr]
    node.profit = gate

    for _, child := range node.child {
        dfs(nodes, child, amount, maxProfit)
        node.profit += (*nodes)[child].profit
    }

    if node.parent != 0 {
        gate /= 2
        (*nodes)[node.parent].profit += gate
        node.profit += gate - amount[node.parent]
    }

    if node.profit > maxProfit {
        maxProfit = node.profit
    }
}

File: data/EXP01/python/2467/2467.output_1.py
# @lc app=leetcode id=2467 lang=python3
from typing import List

class Solution:
    def mostProfitablePath(self, edges: List[List[int]], bob: int, amount: List[int]) -> int:
        adjacency_list = [[] for _ in range(len(amount))]
        for edge in edges:
            adjacency_list[edge[0]].append(edge[1])
            adjacency_list[edge[1]].append(edge[0])
        
        def dfs(node: int, parent: int) -> int:
            total_income = amount[node] // 2 # Alice's share of the gate at the current node
            
            for neighbor in adjacency_list[node]:
                if neighbor != parent: # Avoid going back to the parent node
                    neighbor_income = dfs(neighbor, node)
                    total_income += max(0, neighbor_income) # Only add positive incomes
            
            return total_income
        
        max_income = dfs(0, -1)
        
        return max_income